modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLErrorResponseTestCase.java

```diff
@@ -0,0 +1,306 @@
+            "wso2.is")
+    public void testRelayStateAndACSWithSAMLErrorResponse() throws Exception {
+
+        //create service provider config with mis-matching ACS to generate SAML error response
+        Boolean isAddSuccess = ssoConfigServiceClient
+                .addServiceProvider(createSsoServiceProviderDTO());
+        Assert.assertTrue(isAddSuccess, "Adding a service provider has failed for " + ARTIFACT_ID);
+
+        HttpResponse response;
+        response = sendGetRequest(String.format(SAML_SSO_LOGIN_URL, ARTIFACT_ID, HTTP_POST_BINDING));
+        String samlRequest = extractDataFromResponse(response, "SAMLRequest", 5);
+        Map<String, String> paramters = new HashMap<String, String>();
+        paramters.put("SAMLRequest", samlRequest);
+        paramters.put("RelayState", RELAY_STATE);
+        response = sendSAMLMessage(SAML_SSO_URL, paramters);
+        EntityUtils.consume(response.getEntity());
+        response = sendRedirectRequest(response);
+
+        String sessionKey = extractDataFromResponse(response, "name=\"sessionDataKey\"", 1);
+        response = sendPOSTMessage(sessionKey);
+        EntityUtils.consume(response.getEntity());
+
+        String location = response.getFirstHeader("Location").getValue();
+        Assert.assertTrue(location.contains("RelayState=" + RELAY_STATE), "Redirection header to notification.do" +
+                " page should contain RelayState query param sent with the request");
+        Assert.assertTrue(location.contains("ACSUrl=" + URLEncoder.encode(String.format(INVALID_ACS_URL,
+                ARTIFACT_ID), "UTF-8")), "Redirection header to notification.do" +
+                " page should contain ACS query param");
+    }
+
+
+    private void createApplication() throws Exception {
+
+        ServiceProvider serviceProvider = new ServiceProvider();
+        serviceProvider.setApplicationName(APPLICATION_NAME);
+        serviceProvider.setDescription("This is a test Service Provider");
+        applicationManagementServiceClient.createApplication(serviceProvider);
+        serviceProvider = applicationManagementServiceClient.getApplication(APPLICATION_NAME);
+
+        InboundAuthenticationRequestConfig requestConfig = new InboundAuthenticationRequestConfig();
+        requestConfig.setInboundAuthType(INBOUND_AUTH_TYPE);
+        requestConfig.setInboundAuthKey(ARTIFACT_ID);
+
+        InboundAuthenticationConfig inboundAuthenticationConfig = new InboundAuthenticationConfig();
+        inboundAuthenticationConfig.setInboundAuthenticationRequestConfigs(
+                new InboundAuthenticationRequestConfig[]{requestConfig});
+
+        serviceProvider.setInboundAuthenticationConfig(inboundAuthenticationConfig);
+        applicationManagementServiceClient.updateApplicationData(serviceProvider);
+    }
+
+    private Tomcat getTomcat() {
+
+        Tomcat tomcat = new Tomcat();
+        tomcat.getService().setContainer(tomcat.getEngine());
+        tomcat.setPort(8490);
```

Comment:
There was a constant defined for port value. use that so that this can be centrally managed